sklearn/linear_model/_linear_loss.py

```diff
@@ -287,13 +287,29 @@ def loss_gradient(
 
         if not self.base_loss.is_multiclass:
             grad = np.empty_like(coef, dtype=weights.dtype)
-            grad[:n_features] = X.T @ grad_pointwise + l2_reg_strength * weights
+            with np.errstate(all="raise"):
+                try:
+                    grad[:n_features] = X.T @ grad_pointwise + l2_reg_strength * weights
+                except FloatingPointError:
+                    raise ValueError(
+                        "Overflow detected. Try scaling the target variable or"
+                        " features, or using a different solver"
+                    ) from None
             if self.fit_intercept:
                 grad[-1] = grad_pointwise.sum()
         else:
             grad = np.empty((n_classes, n_dof), dtype=weights.dtype, order="F")
             # grad_pointwise.shape = (n_samples, n_classes)
-            grad[:, :n_features] = grad_pointwise.T @ X + l2_reg_strength * weights
+            with np.errstate(all="raise"):
```

Comment:
If this is an overflow, I think that using `over="raise"` would be better.

Reply:
I tried using ```over="raise"```, however this didn't raise an Exception and the RuntimeWarning appeared instead - when I tested with the dataset I encountered this issue with in the first place https://github.com/scikit-learn/scikit-learn/issues/27016. During one of the iterations, there was an infinity in one the gradients which is where I wanted to raise the Exception.